### ISSUE TITLE
Start-PSBuild failing on Windows

### ISSUE THREAD

**Issue by SteveL-MSFT** [PowerShell/PowerShell#1715]
PS C:\Users\slee\Repos\PowerShell> Start-PSBuild
WARNING: Could not find 'dotnet', appending C:\Users\slee\AppData\Local\Microsoft\dotnet to PATH.
Get-ChildItem : Cannot find path 'C:\Program Files (x86)\Windows Kits\1' because it does not exist.
At C:\Users\slee\Repos\PowerShell\build.psm1:133 char:18
- ...    $mcexe = Get-ChildItem "${env:ProgramFiles(x86)}\Windows Kits\10\" ...
-                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  - CategoryInfo          : ObjectNotFound: (C:\Program Files (x86)\Windows Kits\1:String) [Get-ChildItem], ItemNotF
    oundException
  - FullyQualifiedErrorId : PathNotFound,Microsoft.PowerShell.Commands.GetChildItemCommand

mc.exe not found. Run Start-PSBootstrap or install Microsoft Windows 10 SDK from
https://developer.microsoft.com/en-US/windows/downloads/windows-10-sdk
At C:\Users\slee\Repos\PowerShell\build.psm1:135 char:13
-             throw 'mc.exe not found. Run Start-PSBootstrap or install ...
-             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  - CategoryInfo          : OperationStopped: (mc.exe not foun.../windows-10-sdk:String) [], RuntimeException
  - FullyQualifiedErrorId : mc.exe not found. Run Start-PSBootstrap or install Microsoft Windows 10 SDK from https:/
    /developer.microsoft.com/en-US/windows/downloads/windows-10-sdk
---

Start-PSBootStrap reports that everything needed is installed including Win10 SDK

**Comment by rkeithhill** [comment]
So is Windows 7 not supported for building?  I get an error about `win7-x86` not being supported on a clean Win7 VM I had laying around.

**Comment by rkeithhill** [comment]
OK, using the VC++ build tools I got this far on Windows 7 x64:
![image](https://cloud.githubusercontent.com/assets/5177512/17781612/3ac95b26-652d-11e6-86d3-f107ae35c966.png)

Why would I be getting 401 unauthorized errors?

**Comment by rkeithhill** [comment]
I did a pull to get latest and perhaps coincidentally I'm past the 401 auth errors.  Now it errors out here:

![image](https://cloud.githubusercontent.com/assets/5177512/17788560/805c982e-654b-11e6-9a8a-35666f929f2c.png)

The VC++ build tools did create a VS140COMNTOOLS env var there is a vcvarsall.bat in the VS 14.0 VC dir.

**Comment by ravindrank** [comment]
Regarding cmake, the build process start-psbuild adds the incorrect path (c:\program files (x86)\cmake) to path variable.  On my Windows 10 1511, Cmake gets installed in c:\program files\cmake  (not the x86 pf). Maybe its chocolatey which changed the install path?

**Comment by KarolKaczmarek** [comment]
@rkeithhill 
```no cmake_cxx_compiler could not be found```
suggests you need to add path to cl.exe added to your PATH. 

The previous error you were getting with nuget paths was just temporary and is fixed now, as you've noticed.

and while VC++ tools take 3GB, it's still significant improvement comparing with the size of full VS installation.

@ravindrank I wish there was an easy way to detect what exact path choco installs given component to... that would help get rid of such problems.

That being said, I believe cmake gets added to the PATH automatically during the installation so we could remove that requirement - I need to double check it though.

**Comment by rkeithhill** [comment]
This is the error I find buried in the CMakeError.log for powershell-native:
```
The Windows SDK version 8.1 was not found. Install the required version of Windows SDK ...
```
So it looks like the VC++ build tools don't install everything you need.  Although I though Start-PSBootstrap installed the required SDK on Windows?

**Comment by rkeithhill** [comment]
OK installed the Windows 8.1 SDK on my Windows 7 VM and I get further.  Now it fails here:

![image](https://cloud.githubusercontent.com/assets/5177512/17827392/760aba7e-6639-11e6-9011-0fcde3fd1268.png)

Sorry for the images.  Hyper-V is not letting me copy to the host OS clipboard.

**Comment by mirichmo** [comment]
Fixed by #2568
